### PR TITLE
Changed awsfederatedaccountaccess to use secret reference

### PR DESCRIPTION
**Pull request by rafael-azevedo**
This PR replaces the account name reference with a secret reference in awsfederatedaccountaccess.

### CODE REVIEW

File: deploy/crds/aws_v1alpha1_awsfederatedaccountaccess_crd.yaml [hunk 1]
```diff
@@ -38,20 +38,28 @@ spec:
...
+              - name
+              - namespace
+              type: object
             awsFederatedRoleName:
               description: FederatedRoleName must be the name of a federatedrole cr
                 that currently exists
               type: string
             externalCustomerAWSIAMARN:
-              description: externalCustomerAWSIAMARN holds the external AWS IAM ARN
+              description: ExternalCustomerAWSARN holds the external AWS IAM ARN
```
Comment (fahlmant) [inline deploy/crds/aws_v1alpha1_awsfederatedaccountaccess_crd.yaml]: Where is this change coming from?

Comment (jharrington22) [reply deploy/crds/aws_v1alpha1_awsfederatedaccountaccess_crd.yaml]: +1 keep it lowercase to match actual field.

File: pkg/apis/aws/v1alpha1/awsfederatedaccountaccess_types.go [hunk 2]
```diff
@@ -25,8 +25,8 @@ const (
 type AWSFederatedAccountAccessSpec struct {
 	// ExternalCustomerAWSARN holds the external AWS IAM ARN
 	ExternalCustomerAWSIAMARN string `json:"externalCustomerAWSIAMARN"`
-	// AccountReference holds the name of the associated Account CR to use
-	AccountReference string `json:"accountReference"`
+	// AwsCredentialSecret holds the credentials to the cluster account where the role wil be created
+	AwsCredentialSecret AWSSecretReference `json:"awsCredentialSecret"`
```
Comment (jharrington22) [inline pkg/apis/aws/v1alpha1/awsfederatedaccountaccess_types.go]:
```suggestion
	// AWSCustomerCredentialSecret hold the IAM credentials to the cluster account where the role will be created
	AWSCustomerCredentialSecret AWSSecretReference `json:"awsCredentialSecret"`
```

File: pkg/controller/awsfederatedaccountaccess/awsfederatedaccountaccess_controller.go [hunk 3]
```diff
@@ -147,6 +144,21 @@ func (r *ReconcileAWSFederatedAccountAccess) Reconcile(request reconcile.Request
 		return reconcile.Result{}, err
 	}
 
+	// Get account number of cluster account
+	gciOut, err := awsClient.GetCallerIdentity(&sts.GetCallerIdentityInput{})
+	if err != nil {
+		SetStatuswithCondition(currentFAA, "Failed to get account ID information", awsv1alpha1.AWSFederatedAccountFailed, awsv1alpha1.AWSFederatedAccountStateFailed)
+		reqLogger.Error(ErrFederatedAccessRoleFailedCreate, fmt.Sprintf("Unable to create role requested by '%s'", currentFAA.Name), "AWS ERROR: ", err)
```
Comment (jharrington22) [inline pkg/controller/awsfederatedaccountaccess/awsfederatedaccountaccess_controller.go]: Can we cast this to an AWS error like we do in other places?

**Comment by fahlmant** [comment]
Tested the PR and it creates the role correctly when using the secret. :+1:

**Pull request merged**